### PR TITLE
[chore] Move metrics initialization in service/telemetry

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: service/telemetry/factory.go [hunk 1]
```diff
@@ -7,15 +7,26 @@ import (
...
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configtelemetry"
+	"go.opentelemetry.io/collector/featuregate"
+	"go.opentelemetry.io/collector/service/internal/resource"
 	"go.opentelemetry.io/collector/service/telemetry/internal"
 )
 
+// disableHighCardinalityMetricsfeatureGate is the feature gate that controls whether the collector should enable
+// potentially high cardinality metrics. The gate will be removed when the collector allows for view configuration.
+var disableHighCardinalityMetricsfeatureGate = featuregate.GlobalRegistry().MustRegister(
```
Comment (codeboten) [inline service/telemetry/factory.go:24]: Should this be moved to globalgates instead?

**Review by codeboten — APPROVED** [review]
this looks mostly ok, a big chunk of this code (all the stuff in otelinit) will be removed once https://github.com/open-telemetry/opentelemetry-collector/pull/11079 lands

File: service/telemetry/factory.go (hunk 1 above)
Comment (bogdandrutu) [reply service/telemetry/factory.go:24]: Why? Where else is used? My understanding is that in globalgates we put things shared between components/modules.

Comment (codeboten) [reply service/telemetry/factory.go:24]: ah, i just thought of it as a convenient location for all feature gates 🤣

**Pull request merged**